### PR TITLE
fix: json marsh error for unsupport type: func()

### PR DESCRIPTION
**Pull request by lowang-bh**
split from https://github.com/volcano-sh/volcano/pull/3162

Fixes https://github.com/volcano-sh/volcano/issues/3240

The original issue and background:
#3256

### CODE REVIEW

**Comment by lowang-bh** [comment]
trigger ci

File: pkg/scheduler/api/cluster_info.go [hunk 1]
```diff
@@ -26,8 +26,8 @@ type ClusterInfo struct {
 	Nodes          map[string]*NodeInfo
 	Queues         map[QueueID]*QueueInfo
 	NamespaceInfo  map[NamespaceName]*NamespaceInfo
-	RevocableNodes map[string]*NodeInfo
-	NodeList       []string
+	RevocableNodes map[string]*NodeInfo `json:"-"`
+	NodeList       []string             `json:"-"`
```
Comment (william-wang) [inline pkg/scheduler/api/cluster_info.go:30]: why only change these two variables?

File: pkg/scheduler/api/job_info.go [hunk 2]
```diff
@@ -133,7 +133,7 @@ type TaskInfo struct {
 	Pod        *v1.Pod
 
 	// CustomBindErrHandler is a custom callback func called when task bind err.
-	CustomBindErrHandler func() error
+	CustomBindErrHandler func() error `json:"-"`
```
Comment (william-wang) [inline pkg/scheduler/api/job_info.go:136]: save as above

Comment (lowang-bh) [reply pkg/scheduler/api/job_info.go:136]: func field can not be serialized.

File: pkg/scheduler/api/cluster_info.go (hunk 1 above)
Comment (lowang-bh) [reply pkg/scheduler/api/cluster_info.go:30]: The original PR is to support dump scheduler cache snapshot to json file. Those fields may not concerned by user,  so skip them.

Comment (Monokaix) [reply pkg/scheduler/api/cluster_info.go:30]: I think we just need fix extender marshal err in this pr, and the changes here can be done by original PR and make some comments.

**Review by william-wang — APPROVED** [review]
/approve

**Pull request merged**